include/libtorrent/alert_types.hpp

```diff
@@ -1815,8 +1815,8 @@ namespace libtorrent
 	struct TORRENT_EXPORT add_torrent_alert final : torrent_alert
 	{
 		// internal
-		add_torrent_alert(aux::stack_allocator& alloc, torrent_handle h
-			, add_torrent_params const& p, error_code ec);
+		add_torrent_alert(aux::stack_allocator& alloc, torrent_handle const& h
```

Comment:
it's still a little obscure to me the "move semantics", not clear the exact rules of how/when/why the default constructors and operators are generated. Other point that keep me thinking is what happens when the object is more heavy in the stack than in the heap.